src/misclibs.F

```diff
@@ -4293,25 +4106,19 @@ subroutine calcener(nstat,rstat,ruh,rvh,zh,rmh,pi0,th0,rho,ua,va,wa,ppi,tha,   &
       ep=0.0d0
       le=0.0d0
       !$acc data create(foo1,foo2,foo3,foo4)
-#ifdef _B4B
-      !$acc update host(ruh,rvh,rmh,ua,va,wa,qv,ql,qi,rho,vr,pi0,ppi)
-#else
-      !$acc parallel default(present) private(i,j,k,u,v,w,tmp,qtot,tem) &
-      !$acc reduction(+:foo1t,foo2t,foo3t,foo4t,ek,ei,ep,le)
-#endif
       !$omp parallel do default(shared)  &
       !$omp private(i,j,k,u,v,w,tmp,qtot,tem)
-#ifndef _B4B
-      !$acc loop gang 
-#endif
+      !$acc parallel default(present) private(i,j,k,u,v,w,tmp,qtot,tem) &
+      !$acc reduction(+:foo1t,foo2t,foo3t,foo4t,ek,ei,ep,le)
+      !JMD internal compiler error
```

Comment:
Unfortunately, the addition of a reduction() clause triggered an internal compiler error. So I reverted it to the previous loop structure

Reply:
Thanks for reporting the internal compiler error. No idea about why it happens. After a second thought, I think it may be safer to write the reductions here as a separate acc parallel region. My main concern is that there is no guarantee for the first k loop to be completed before we execute the second k loop for reduction (I could be wrong though).